pkg/hints/math_hints.go

```diff
@@ -151,3 +154,253 @@ func sqrt(ids IdsManager, vm *VirtualMachine) error {
 	ids.Insert("root", NewMaybeRelocatableFelt(root_felt), vm)
 	return nil
 }
+
+func assertLeFelt(ids IdsManager, vm *VirtualMachine, scopes *ExecutionScopes, constants *map[string]Felt) error {
+	// Fetch constants
+	primeOver3HighFelt, err := ids.GetConst("PRIME_OVER_3_HIGH", constants)
+	if err != nil {
+		return err
+	}
+	primeOver3High := primeOver3HighFelt.ToBigInt()
+	primeOver2HighFelt, err := ids.GetConst("PRIME_OVER_2_HIGH", constants)
+	if err != nil {
+		return err
+	}
+	primeOver2High := primeOver2HighFelt.ToBigInt()
+	// Fetch ids variables
+	aFelt, err := ids.GetFelt("a", vm)
+	if err != nil {
+		return err
+	}
+	a := aFelt.ToBigInt()
+	bFelt, err := ids.GetFelt("b", vm)
+	if err != nil {
+		return err
+	}
+	b := bFelt.ToBigInt()
+	rangeCheckPtr, err := ids.GetRelocatable("range_check_ptr", vm)
+	if err != nil {
+		return err
+	}
+	// Hint Logic
+	cairoPrime, _ := new(big.Int).SetString(CAIRO_PRIME_HEX, 0)
```

Comment:
why the base of `SetString` is 0? should not be 16 if converting an hex value?

Reply:
`CAIRO_PRIME_HEX` starts with "0x", and `SetString` uses the prefix to determine the base if base is 0